contracts/token/common/ERC2981/ERC2981Internal.sol

```diff
@@ -0,0 +1,40 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.8;
+
+import { ERC2981Storage } from './ERC2981Storage.sol';
+import { IERC2981Internal } from './IERC2981Internal.sol';
+
+/**
+ * @title ERC2981 internal functions
+ */
+abstract contract ERC2981Internal is IERC2981Internal {
+    using ERC2981Storage for ERC2981Storage.Layout;
+
+    function _royaltyInfo(uint256 tokenId, uint256 salePrice)
+        internal
+        view
+        virtual
+        returns (address, uint256)
+    {
+        ERC2981Storage.Layout storage l = ERC2981Storage.layout();
+
+        uint256 royalty = l.royalties[tokenId] > 0
+            ? l.royalties[tokenId]
+            : l.royalty;
+
+        if (0 == royalty) {
+            revert ERC2981Internal__RoyaltyNotSet();
+        }
```

Comment:
This seems like a valid case, especially in a contract where royalties are set only on a per-token basis.  Is this revert case mentioned in the EIP?

Reply:
The revert case is not mentioned in the EIP. Now that you point it out, it probably makes more sense to allow the royalty to be 0 rather than requiring the global or local royalty to be set. If that's the case, it may be necessary to use either the global or local royalty. This can be instrumented by using a flag, if its true use global, false, use local.

```js
function _getRoyaltyBPS(ERC2981Storage.Layout storage l, uint256 tokenId)
    internal
    view
    virtual
    returns (uint256)
{
    return l.DEFAULT_GLOBAL ? l.royalty : l.royalties[tokenId];
}
```

Reply:
No, it's fine just like you have it:
```solidity
uint256 royalty = l.royalties[tokenId] > 0
            ? l.royalties[tokenId]
            : l.royalty;
```
Only issue there is that the mapping is queried twice - better to store `localRoyalty` or something in a local variable and then check if it's 0.

Reply:
As for the revert case, let's just delete it.